graphium/data/datamodule.py

```diff
@@ -670,12 +671,17 @@ def __init__(
             weights_col: The column name of the weights
             weights_type: The type of weights
             idx_col: The column name of the indices
+            mol_ids_col: The column name of the molecule ids
             sample_size: The size of the sample
             split_val: The fraction of the data to use for validation
             split_test: The fraction of the data to use for testing
             seed: The seed to use for the splits and subsampling
             splits_path: The path to the splits
         """
+
+        if df is None:
+            assert df_path is not None, "Either df or df_path must be provided"
```

Comment:
`assert` can officially be optimized out when compiling Python and should thus only be used for checking programming errors, but not user errors. I would thus raise an error here.

Reply:
For practicality sake, I will change this here. Don't worry about it.